corehq/apps/es/management/commands/ensure_indices_reindexed.py

```diff
@@ -0,0 +1,165 @@
+from django.core.management.base import BaseCommand, CommandError
+
+from corehq.apps.es import CANONICAL_NAME_ADAPTER_MAP, const
+from corehq.apps.es.client import manager
+from corehq.apps.es.transient_util import doc_adapter_from_cname
+
+helptext = """
+The management command is intended for execution during Django migrations and is designed for use
+in Elasticsearch upgrades. Its purpose is to ensure the reindexing of all Elasticsearch indexes.
+The command accomplishes this by comparing the document counts in both primary and secondary indexes for each app.
+If the difference in count is not zero for indexes or exceeds 1% for high-frequency indices such as
+case-search, cases, and forms, the command will fail.
+However, it will skip execution if the current Elasticsearch version is greater than the specified version
+which means that Elasticsearch is already upgraded.
+"""
+
+
+class Command(BaseCommand):
+
+    help = helptext
+
+    REASONABLE_DIFF_PERCENT = 1
+
+    def add_arguments(self, parser):
+        parser.add_argument(
+            '--target_es_version', type=int, choices=[1, 2, 5], required=True,
```

Comment:
What do you think of renaming this to `minimum_es_version`? When I read `target_es_version` I thought it was referencing the version that the reindex was intended to support, not the current or minimum supported ES version.

Reply:
I spent some time while thinking about this variable name and have named it `minimum_es_version` earlier and then changed it to `target_es_version` before putting up the PR  😅.  `minimum` would mean the command can run on versions greater that the version specified which is not true.
My reasoning to use `target_es_version` was that the command would run only on the targetting ES version. 

I think calling it `only_run_for` would make it more clear that the command will run only on the specified ES version.